packages/medusa/src/api-v2/admin/collections/validators.ts

```diff
@@ -29,13 +29,19 @@ export const AdminGetCollectionsParams = createFindParams({
 export type AdminCreateCollectionType = z.infer<typeof AdminCreateCollection>
 export const AdminCreateCollection = z.object({
   title: z.string(),
-  handle: z.string().optional(),
+  handle: z
+    .string()
+    .regex(/^[a-z0-9]+(?:-[a-z0-9]+)*$/)
+    .optional(),
```

Comment:
sugg: we have a `utils/common-validators/` folder to add common validators where you can add this, I'd assume this might be needed for categories and other `handle` requiring endpoints as well.